File: UserManagment/UserManagment.Domain/Entities/TutorProfile.cs
using UserManagment.Domain.Enums;

namespace UserManagment.Domain.Entities;

public class TutorProfile
{
    public Guid Id { get; set; }
    public Guid UserAccountId { get; set; }

    public string ProfilePhotoUrl { get; set; } = string.Empty;
    public string HighestDegree { get; set; } = string.Empty;
    public string FieldOfStudy { get; set; } = string.Empty;
    public string InstitutionName { get; set; } = string.Empty;
    public int GraduationYear { get; set; }
    public string DegreeCertificateUrl { get; set; } = string.Empty;

    public string SubjectsCsv { get; set; } = string.Empty;
    public string GradeLevelsCsv { get; set; } = string.Empty;
    public int YearsOfExperience { get; set; }
    public string LanguagesCsv { get; set; } = string.Empty;

    public TeachingMode TeachingMode { get; set; }
    public string InPersonLocation { get; set; } = string.Empty;
    public decimal HourlyFee { get; set; }
    public decimal? MonthlyFee { get; set; }
    public string AvailableDaysCsv { get; set; } = string.Empty;
    public string AvailableTimeSlotsCsv { get; set; } = string.Empty;
    public string UnavailableDatesCsv { get; set; } = string.Empty;

    public string Bio { get; set; } = string.Empty;
    public string TeachingMethodology { get; set; } = string.Empty;
    public string Achievements { get; set; } = string.Empty;

    public GovernmentIdType GovernmentIdType { get; set; }
    public string GovernmentIdDocumentUrl { get; set; } = string.Empty;
    public bool BackgroundCheckConsent { get; set; }
    public string TeachingLicensesOrCertificatesUrl { get; set; } = string.Empty;

    public bool TermsAccepted { get; set; }
    public bool PrivacyAccepted { get; set; }
    public bool CommissionPolicyAccepted { get; set; }

    public TutorVerificationStatus VerificationStatus { get; set; } = TutorVerificationStatus.Pending;
    public string? VerificationNotes { get; set; }
    public string? ReviewedByAdminId { get; set; }
    public DateTime? ReviewedAtUtc { get; set; }

    public int ProfileCompleteness { get; set; }
    public double? AverageRating { get; set; }
    public int ReviewCount { get; set; } = 0;
    public DateTime CreatedAtUtc { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAtUtc { get; set; } = DateTime.UtcNow;

    public UserAccount UserAccount { get; set; } = null!;

    public int CalculateProfileCompleteness()
    {
        var checks = new[]
        {
            !string.IsNullOrWhiteSpace(ProfilePhotoUrl),
            !string.IsNullOrWhiteSpace(HighestDegree),
            !string.IsNullOrWhiteSpace(FieldOfStudy),
            !string.IsNullOrWhiteSpace(InstitutionName),
            GraduationYear > 0,
            !string.IsNullOrWhiteSpace(DegreeCertificateUrl),
            !string.IsNullOrWhiteSpace(SubjectsCsv),
            !string.IsNullOrWhiteSpace(GradeLevelsCsv),
            YearsOfExperience >= 0,
            !string.IsNullOrWhiteSpace(LanguagesCsv),
            !string.IsNullOrWhiteSpace(AvailableDaysCsv),
            !string.IsNullOrWhiteSpace(AvailableTimeSlotsCsv),
            HourlyFee > 0,
            !string.IsNullOrWhiteSpace(Bio),
            !string.IsNullOrWhiteSpace(TeachingMethodology),
            !string.IsNullOrWhiteSpace(GovernmentIdDocumentUrl),
            BackgroundCheckConsent,
            TermsAccepted,
            PrivacyAccepted,
            CommissionPolicyAccepted
        };

        var completed = checks.Count(value => value);
        return (int)Math.Round((completed / (double)checks.Length) * 100, MidpointRounding.AwayFromZero);
    }
}

File: tutor-service/src/main/java/com/mentorhub/tutorservice/model/TutorProfile.java
package com.mentorhub.tutorservice.model;

import jakarta.persistence.*;
import lombok.*;
import java.math.BigDecimal;
import java.time.Instant;
import java.util.UUID;

@Entity
@Table(name = "tutor_profiles")
@Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder
public class TutorProfile {
    @Id
    @Column(name = "\"Id\"")
    private UUID id;

    @Column(name = "\"UserAccountId\"", nullable = false)
    private UUID userAccountId;

    @Column(name = "\"ProfilePhotoUrl\"")
    private String profilePhotoUrl;

    @Column(name = "\"HighestDegree\"", length = 200)
    private String highestDegree;

    @Column(name = "\"FieldOfStudy\"", length = 200)
    private String fieldOfStudy;

    @Column(name = "\"InstitutionName\"", length = 240)
    private String institutionName;

    @Column(name = "\"GraduationYear\"")
    private Integer graduationYear;

    @Column(name = "\"DegreeCertificateUrl\"")
    private String degreeCertificateUrl;

    @Column(name = "\"SubjectsCsv\"")
    private String subjectsCsv;

    @Column(name = "\"GradeLevelsCsv\"")
    private String gradeLevelsCsv;

    @Column(name = "\"YearsOfExperience\"")
    private Integer yearsOfExperience;

    @Column(name = "\"LanguagesCsv\"")
    private String languagesCsv;

    @Column(name = "\"TeachingMode\"")
    private Integer teachingMode;

    @Column(name = "\"InPersonLocation\"", length = 240)
    private String inPersonLocation;

    @Column(name = "\"HourlyFee\"", precision = 12, scale = 2)
    private BigDecimal hourlyFee;

    @Column(name = "\"MonthlyFee\"", precision = 12, scale = 2)
    private BigDecimal monthlyFee;

    @Column(name = "\"AvailableDaysCsv\"")
    private String availableDaysCsv;

    @Column(name = "\"AvailableTimeSlotsCsv\"")
    private String availableTimeSlotsCsv;

    @Column(name = "\"Bio\"")
    private String bio;

    @Column(name = "\"TeachingMethodology\"")
    private String teachingMethodology;

    @Column(name = "\"Achievements\"")
    private String achievements;

    @Column(name = "\"GovernmentIdType\"")
    private Integer governmentIdType;

    @Column(name = "\"GovernmentIdDocumentUrl\"")
    private String governmentIdDocumentUrl;

    @Column(name = "\"BackgroundCheckConsent\"")
    private Boolean backgroundCheckConsent;

    @Column(name = "\"TeachingLicensesOrCertificatesUrl\"")
    private String teachingLicensesOrCertificatesUrl;

    @Column(name = "\"TermsAccepted\"")
    private Boolean termsAccepted;

    @Column(name = "\"PrivacyAccepted\"")
    private Boolean privacyAccepted;

    @Column(name = "\"CommissionPolicyAccepted\"")
    private Boolean commissionPolicyAccepted;

    @Column(name = "\"VerificationStatus\"")
    private Integer verificationStatus;

    @Column(name = "\"VerificationNotes\"", length = 1500)
    private String verificationNotes;

    @Column(name = "\"ReviewedByAdminId\"", length = 120)
    private String reviewedByAdminId;

    @Column(name = "\"ReviewedAtUtc\"")
    private Instant reviewedAtUtc;

    @Column(name = "\"ProfileCompleteness\"")
    private Integer profileCompleteness;

    @Column(name = "\"AverageRating\"")
    private Double averageRating;

    @Column(name = "\"ReviewCount\"")
    private Integer reviewCount;

    @Column(name = "\"CreatedAtUtc\"", nullable = false)
    private Instant createdAtUtc;

    @Column(name = "\"UpdatedAtUtc\"")
    private Instant updatedAtUtc;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "\"UserAccountId\"", insertable = false, updatable = false)
    private UserAccount userAccount;
}
